kale/embed/video_i3d.py

```diff
@@ -436,10 +480,29 @@ def extract_features(self, x):
         return self.avg_pool(x)
 
 
-def i3d(name, num_channels, pretrained=False, progress=True):
-    model = InceptionI3d(in_channels=num_channels)
+def i3d(name, num_channels, num_classes, pretrained=False, progress=True):
+    """Get InceptionI3d module w/o pretrained model."""
+    model = InceptionI3d(in_channels=num_channels, num_classes=num_classes)
 
     if pretrained:
         state_dict = load_state_dict_from_url(model_urls[name], progress=progress)
+        # delete logits.conv3d parameters due to different class number.
+        # state_dict.pop("logits.conv3d.weight")
+        # state_dict.pop("logits.conv3d.bias")
+        # model.load_state_dict(state_dict, strict=False)
```

Comment:
Are the above useful to keep? Options/alternatives?

Reply:
The pre-trained model has 400 categories, while we have 6 to 8. Loading parameter to the model by `model.load_state_dict(state_dict)` requires that parameter and model have exactly the same dimensions (No difference, no missing). However, the output of the last layer for us is 6 to 8,  not 400. There are two solutions.

1. Delete the last layer's pre-trained parameter and only load the parameters before the last. This raises an issue about missing the last layer's parameter. Thus, I set `strict=False` in `model.load_state_dict` to ignore this missing issue following some others' codes. These commented codes are doing this. However, there is a risk that I won’t be alerted if other layers go wrong.

2. In our setting, we have our own classifier and only use the output of average pooling in I3D. This average pooling layer is before the last layer. Therefore, I choose to load the pre-trained model without changing the last layer but don't train the last layer in training. This is much safer. 

This is the reason why I comment these lines and fix `output_channels` to 400. I don't delete them because we may use the original I3D to train something in the future, and they will be useful.

Reply:
I am not sure how someone other than you could figure it out with just the comment "delete logits.conv3d parameters due to different class number." You should review your code from a user's perspective, not yours, just as when you write a paper.